client_test.go

```diff
@@ -56,7 +50,44 @@ func TestMakeRequest(t *testing.T) {
 		req := client.newRequest("GET", "/v2/foobar")
 		resp, err := client.doRequest(req)
 		So(err, ShouldBeNil)
-		So(client.GetToken(), ShouldEqual, "bearer foobar")
 		So(resp, ShouldNotBeNil)
 	})
 }
+
+func TestTokenRefresh(t *testing.T) {
+	gomega.RegisterTestingT(t)
+	Convey("Test making request", t, func() {
+		server := fakeUAAServer()
+		c := &Config{
+			ApiAddress:   server.URL,
+			LoginAddress: server.URL,
+			Username:     "foo",
+			Password:     "bar",
+		}
+		client := NewClient(c)
+		gomega.Consistently(client.GetToken()).Should(gomega.Equal("bearer foobar2"))
+		gomega.Eventually(client.GetToken(), "3s").Should(gomega.Equal("bearer foobar3"))
+	})
+}
+
+func fakeUAAServer() *httptest.Server {
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+	m := martini.New()
+	m.Use(render.Renderer())
+	r := martini.NewRouter()
+	count := 1
+	r.Post("/oauth/token", func(r render.Render) {
+		r.JSON(200, map[string]interface{}{
+			"token_type":    "bearer",
+			"access_token":  "foobar" + strconv.Itoa(count),
+			"refresh_token": "barfoo",
+			"expires_in":    3,
```

Comment:
with regard to the `expires_in` time, the token will be seen as expired before the 3 seconds have elapsed. Since an [expiryDelta](https://github.com/golang/oauth2/blob/master/token.go#L101) is taken into account.